docs/tutorials/schrödinger_equation.rst

```diff
@@ -0,0 +1,271 @@
+We define the initial state, the time span for the simulation, and time point we want to simulate.
+
+.. jupyter-execute::
+
+    init_state = np.array([1. + 0j,0. + 0j])
+
+    t_span = [0,(N-1) * dt]
+    T = np.linspace(0,(N-1) * dt,N)
+
+We solve by using ``scipy.integrate.solve_ivp`` and plot the probabilities of each state.
+
+.. jupyter-execute::
+
+    sol = solve_ivp(rhs,t_span,init_state,method='RK45',t_eval=T)
+    probabilities = state_probabilities(sol.y)
+
+    plt.plot(sol.t, probabilities[0], label="0")
+    plt.plot(sol.t, probabilities[1], label="1")
+    plt.xlabel('T')
+    plt.ylabel('Probability')
+    plt.legend()
+    plt.show()
+
+
+Second, we solve the equation by using Jax.array as the input and ``jax.experimental.ode.odeint`` as a solver.
+
+.. jupyter-execute::
+
+    init_state = jnp.array([1. + 0j,0. + 0j])
+
+    t_span = [0,(N-1) * dt]
+    T = jnp.linspace(0,(N-1) * dt,N)
+
+    sol = odeint(lambda y,t: rhs(t,y), init_state, T)
+    probabilities = state_probabilities(sol.T)
+    plt.plot(T, probabilities[0], label="0")
+    plt.plot(T, probabilities[1], label="1")
+    plt.xlabel('T')
+    plt.ylabel('Probability')
+    plt.legend()
+    plt.show()
+
+
+
+4. Register custom solvers using :meth:`.Alias.register_function`
+-----------------------------------------------------------------
+
+In this section, we introduce how to register the function, taking the 4th order Runge-Kutta method
+to solve ordinary differential equations as an example.
+
+The 4th order Runge-Kutta fomula is here:
+
+.. math::
+
+    k_1 &= h \cdot f(t_n, y_n) \\
+    k_2 &= h \cdot f(t_n + \frac{h}{2}, y_n + \frac{k_1}{2}) \\
+    k_3 &= h \cdot f(t_n + \frac{h}{2}, y_n + \frac{k_2}{2}) \\
+    k_4 &= h \cdot f(t_n + h, y_n + k_3) \\
+    y_{n+1} &= y_n + \frac{1}{6}(k_1 + 2k_2 + 2k_3 + k_4)
+
+, where :math:`y_{n}`, :math:`t_{n}`, and :math:`h` are current solution, current time, and time step size, respectively.
+
+We define the Runge-Kutta method to be used later here:
+
+.. jupyter-execute::
+
+    def runge_kutta_step(n, state):
+        k1 = dt * rhs(n * dt, state)
+        k2 = dt * rhs(n * dt + 0.5 * dt, state + 0.5*k1)
+        k3 = dt * rhs(n * dt + 0.5 * dt, state + 0.5*k2)
+        k4 = dt * rhs(n * dt + dt, state + k3)
+        return (k1 + 2*k2 + 2*k3 + k4) / 6.
+
+When writing the custom solver, we write and register a version for both NumPy and JAX. 
+The NumPy version uses standard Python loops, whereas the JAX version uses the JAX looping construct ``jax.lax.scan``.
+
+In Numpy case, we define the function ``runge_kutta`` for Numpy.
+
+.. jupyter-execute::
+
+    @alias.register_function(lib="numpy", path="runge_kutta")
+    def _(state, N):
+        probabilities = []
+        for n in range(N):
+            probabilities.append(state_probabilities(state))
+            state+= runge_kutta_step(n, state)
+        return probabilities
```

Comment:
In line with the above comment about defining `runge_kutta_step` to have signature `runge_kutta_step(y, t, dt, rhs)`, I think this could be rewritten to have a signature like

```
def _(y0, t0, dt, N, rhs):
   ....
```
It's not a huge deal but this is the more common way in which the interface for DE solvers are written - an argument for the initial state, some way of specifying the solution interval, and the passed RHS function.

Reply:
I changed at https://github.com/Qiskit-Extensions/arraylias/pull/25/commits/a5e84316f9712bcb652813d1494a97185cfd8d76.